sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/AsynReadWithMultipleClients.java

```diff
@@ -0,0 +1,299 @@
+                    AsynReadWithMultipleClients.this.onSuccess();
+
+                    synchronized (count) {
+                        count.incrementAndGet();
+                        count.notify();
+                    }
+                }
+
+                @Override
+                protected void hookOnError(Throwable throwable) {
+                    failureMeter.mark();
+                    logger.error("Encountered failure {} on thread {}",
+                        throwable.getMessage(), Thread.currentThread().getName(), throwable);
+                    concurrencyControlSemaphore.release();
+                    AsynReadWithMultipleClients.this.onError(throwable);
+
+                    synchronized (count) {
+                        count.incrementAndGet();
+                        count.notify();
+                    }
+                }
+            };
+
+            performWorkload(baseSubscriber, i);
+        }
+        synchronized (count) {
+            while (count.get() < i) {
+                count.wait();
+            }
+        }
+
+        long endTime = System.currentTimeMillis();
+        logger.info("[{}] operations performed in [{}] seconds.",
+            configuration.getNumberOfOperations(), (int) ((endTime - startTime) / 1000));
+
+        reporter.report();
+        reporter.close();
+    }
+
+    void shutdown() {
+        for (CosmosAsyncClient asyncClient : clientDocsMap.keySet()) {
+            asyncClient.close();
+        }
+    }
+
+    protected void onSuccess() {
+    }
+
+    protected void onError(Throwable throwable) {
+    }
+
+    private PojoizedJson generateDocument(String idString, String dataFieldValue, String partitionKey) {
+        com.azure.cosmos.benchmark.PojoizedJson instance = new com.azure.cosmos.benchmark.PojoizedJson();
+        Map<String, String> properties = instance.getInstance();
+        properties.put("id", idString);
+        properties.put(partitionKey, idString);
+
+        for (int i = 0; i < configuration.getDocumentDataFieldCount(); i++) {
+            properties.put("dataField" + i, dataFieldValue);
+        }
+
+        return instance;
+    }
+
+    private boolean shouldContinue(long startTimeMillis, long iterationCount) {
+
+        Duration maxDurationTime = configuration.getMaxRunningTimeDuration();
+        int maxNumberOfOperations = configuration.getNumberOfOperations();
+
+        if (maxDurationTime == null) {
+            return iterationCount < maxNumberOfOperations;
+        }
+
+        if (startTimeMillis + maxDurationTime.toMillis() < System.currentTimeMillis()) {
+            return false;
+        }
+
+        if (maxNumberOfOperations < 0) {
+            return true;
+        }
+
+        return iterationCount < maxNumberOfOperations;
+    }
+
+    private void createClients() {
+        String csvFile = "clientHostAndKey.txt";
+        BufferedReader br = null;
+        String line = "";
+        String splitBy = ",";
+        try {
```

Comment:
please use try-with-resources to simplify the try/catch pattern:
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Reply:
done